systemverilog-plugin/UhdmAst.cc

```diff
@@ -502,11 +509,23 @@ static void convert_packed_unpacked_range(AST::AstNode *wire_node)
         return false;
     }();
     if (convert_node) {
+        // if not already converted
         if (wire_node->multirange_dimensions.empty()) {
-            packed_size = add_multirange_attribute(wire_node, packed_ranges);
-            unpacked_size = add_multirange_attribute(wire_node, unpacked_ranges);
-            size = packed_size * unpacked_size;
-            ranges.push_back(make_range(size - 1, 0));
+            const size_t packed_size = add_multirange_attribute(wire_node, packed_ranges);
+            const size_t unpacked_size = add_multirange_attribute(wire_node, unpacked_ranges);
+            if (packed_ranges.size() == 1 && unpacked_ranges.empty()) {
+                ranges.push_back(packed_ranges[0]->clone());
+            } else if (unpacked_ranges.size() == 1 && packed_ranges.empty()) {
+                ranges.push_back(unpacked_ranges[0]->clone());
+            } else {
+                // currently we have limited support
+                // for multirange wires that doesn't start from 0
+                warn_start_range(packed_ranges);
+                warn_start_range(unpacked_ranges);
```

Comment:
Shouldn't this be an error? In such cases results are invalid.
I did test only the snippet posted above, so maybe there are cases where the warning is emitted and the result is correct.

Reply:
There are cases where result is almost correct, e.g.:
```systemverilog
  module dut();
  parameter int unsigned DataWidth         = 32;
  localparam int unsigned NUM_WORDS  = 10;
  logic [NUM_WORDS-1:1][DataWidth-1:0] rf_reg_q;
  
  for (genvar i = 1; i < NUM_WORDS; i++) begin : g_rf_flops
    assign rf_reg_q[i] = i;
  end
  
  endmodule
 ```
 produces:
 ```verilog
 module dut();

  wire [287:0] rf_reg_q;
  assign rf_reg_q[31:0] = 32'd1;
  assign rf_reg_q[63:32] = 32'd2;
  assign rf_reg_q[95:64] = 32'd3;
  assign rf_reg_q[127:96] = 32'd4;
  assign rf_reg_q[159:128] = 32'd5;
  assign rf_reg_q[191:160] = 32'd6;
  assign rf_reg_q[223:192] = 32'd7;
  assign rf_reg_q[255:224] = 32'd8;
  assign rf_reg_q[287:256] = 32'd9;
endmodule
```
and this case is used in ``opentitan``.
Promoting this warning to error would force us to fix usage of multiranges that doesn't start from 0 and this could take some time. I don't think it is worth doing this now as it is not used very often.

WDYT?

Reply:
We decided offline that it is ok to warn for now.